src/auditlog/models.py

```diff
@@ -94,6 +94,9 @@ def get_for_objects(self, queryset):
 
         if isinstance(primary_keys[0], integer_types):
             return self.filter(content_type=content_type).filter(Q(object_id__in=primary_keys)).distinct()
+        elif isinstance(queryset.model._meta.pk, models.UUIDField):
```

Comment:
Right. Apparently I wrongly assumed it would evaluate the whole list.

I would propose just evaluating it - we need to do it at some point and that would make the original if statement a bit faster AND keep everything very readable. (And we don't have the issue mentioned by @audiolion about the explicit field name.)

In any case, since we are concerned with the type of the value it would be better to check the type of the value rather than the type of something else that provided the value.